### PR TITLE
harmonist: init at 0.1

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

To add a new roguelike game.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

:alien:

### CODE REVIEW

**Review by JohnAZoidberg — APPROVED** [review]
Plays fine. Saving and loading games also works.

**Comment by etu** [comment]
@GrahamcOfBorg build harmonist

File: pkgs/games/harmonist/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{stdenv, fetchurl, buildGoPackage}:
+
+buildGoPackage rec {
+
+  name = "harmonist-${version}";
```
Comment (etu) [inline pkgs/games/harmonist/default.nix]: Can be replaced by `pname = "harmonist";`. Then name will be built by itself.

File: pkgs/games/harmonist/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    description = "A stealth coffee-break roguelike game";
+    longDescription = ''
+      Harmonist is a stealth coffee-break roguelike game. The game has a heavy
+      focus on tactical positioning, light and noise mechanisms, making use of
+      various terrain types and cones of view for monsters. Aiming for a
+      replayable streamlined experience, the game avoids complex inventory
+      management and character building, relying on items and player
+      adaptability for character progression.
+    '';
+    homepage = https://harmonist.tuxfamily.org/;
```
[the remark this answers is not in the document]
Comment (etu) [reply pkgs/games/harmonist/default.nix]: If you do the change from name to pname, you can update this as well.

**Review by etu — APPROVED** [review]
Looks good overall, it would be great if you could look into the notes left by me and @JohnAZoidberg and then I have no problem with merging this.

**Comment by freepotion** [comment]
@etu @JohnAZoidberg  It's done. :alien:

File: pkgs/games/harmonist/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{stdenv, fetchurl, buildGoPackage}:
+
+buildGoPackage rec {
+
+  pname = "harmonist-${version}";
```
Comment (etu) [inline pkgs/games/harmonist/default.nix]: pname shouldn't contain version, that's the point with pname, it's used to build name by combining pname with version

Comment (freepotion) [reply pkgs/games/harmonist/default.nix]: Thanks, I didn't find it in the documentation. Fixed. I hope it's all right now. Do I understand correctly that a name is optional if a pname is defined?

**Comment by etu** [comment]
@freepotion Thanks! :)

File: pkgs/games/harmonist/default.nix (hunk 3 above)
Comment (etu) [reply pkgs/games/harmonist/default.nix]: It was a convention in the python ecosystem within nix that has spread to stdenv.mkDerivation and can be used in most places. Not mandatory but nice :)

**Comment by freepotion** [comment]
@etu Thank you too! :)

File: pkgs/games/harmonist/default.nix (hunk 3 above)
Comment (freepotion) [reply pkgs/games/harmonist/default.nix]: Oh, I got it. Thanks :)

**Pull request merged**